spec/acceptance/dataset_items_spec.rb

```diff
@@ -892,4 +893,104 @@ def body_params
 
   end
 
+
+  ################################
+  # FILTER TODO
+  ################################
+
+  context 'filter todo' do
+
+    # with deep route including dataset id
+    # One item has a different dataset id, so only 6 items
+    get '/datasets/:dataset_id/dataset_items/filter_todo' do
+      dataset_id_param
+      let(:dataset_id) { dataset.id }
+      let(:authentication_token) { admin_token }
+      standard_request_options(
+          :get,
+          'FILTER TODO (as admin)',
+          :ok,
+          {
+              response_body_content: ['"start_time_seconds":11.0'],
+              expected_json_path: 'data/0/start_time_seconds',
+              data_item_count: 6
+          }
+      )
+    end
+
+    # permissions will be the same for reader,writer,owner so they will have
+    # the same response for the same filter params.
+    # Using nested path, which will filter out the no access item and also the
+    # item from a different dataset, leaving 5 dataset items
+    regular_user_opts = {
+        response_body_content: ['"start_time_seconds":11.0'],
+        expected_json_path: 'data/0/start_time_seconds',
+        data_item_count: 5
+    }
+
+    get '/datasets/:dataset_id/dataset_items/filter_todo' do
+      let(:dataset_id) { dataset.id }
+      let(:authentication_token) { owner_token }
+      standard_request_options(:get,'FILTER TODO (as owner)',:ok,regular_user_opts)
+    end
+
+    get '/datasets/:dataset_id/dataset_items/filter_todo' do
+      let(:dataset_id) { dataset.id }
+      let(:authentication_token) { writer_token }
+      standard_request_options(:get,'FILTER TODO (as writer)',:ok,regular_user_opts)
+    end
+
+    get '/datasets/:dataset_id/dataset_items/filter_todo' do
+      let(:dataset_id) { dataset.id }
+      let(:authentication_token) { reader_token }
+      standard_request_options(:get,'FILTER TODO (as reader)',:ok,regular_user_opts)
+    end
+
+    get '/datasets/:dataset_id/dataset_items/filter_todo' do
+      let(:dataset_id) { dataset.id }
+      let(:authentication_token) { no_access_token }
+      standard_request_options(
+          :get,
+          'FILTER TODO (as no access)',
+          :ok,
+          {response_body_content: ['"order_by":"priority"'], expected_json_path: 'data', data_item_count: 0}
+      )
+    end
+
+    get '/datasets/:dataset_id/dataset_items/filter_todo' do
+      let(:dataset_id) { dataset.id }
+      let(:authentication_token) { invalid_token }
+      standard_request_options(
+          :get,
+          'FILTER TODO (as invalid token)',
+          :unauthorized,
+          {expected_json_path: get_json_error_path(:sign_up)}
+      )
+    end
+
+    # not logged in users can filter dataset items, but they won't get any items that they don't have permission for
+    get '/datasets/:dataset_id/dataset_items/filter_todo' do
```

Comment:
is it worth adding a test case for public projects? I.e. a non-logged in user may have access to `audio_recordings` and `dataset_items`